fs-objects.go

```diff
@@ -143,7 +145,73 @@ func (fs fsObjects) GetObjectInfo(bucket, object string) (ObjectInfo, error) {
 
 // PutObject - create an object.
 func (fs fsObjects) PutObject(bucket string, object string, size int64, data io.Reader, metadata map[string]string) (string, error) {
-	return putObjectCommon(fs.storage, bucket, object, size, data, metadata)
+	// Verify if bucket is valid.
+	if !IsValidBucketName(bucket) {
+		return "", BucketNameInvalid{Bucket: bucket}
+	}
+	// Check whether the bucket exists.
+	if !isBucketExist(fs.storage, bucket) {
+		return "", BucketNotFound{Bucket: bucket}
+	}
+	if !IsValidObjectName(object) {
+		return "", ObjectNameInvalid{
+			Bucket: bucket,
+			Object: object,
+		}
+	}
+
+	fileWriter, err := fs.storage.CreateFile(bucket, object)
```

Comment:
This is incorrect, even FS fs-objects is supposed to use temporary location and rename.  Otherwise there is no cleaner mechanism to clean temporary files on FS which would lie dormant into the future. 

That was the whole reason why we use `tmp` even for FS, so you have to  converge this back and for XL do what is necessary additionally in its own implementation.